ibis/udf/vectorized.py

```diff
@@ -102,6 +52,10 @@ def func(*args):
 
 
 def _udf_decorator(node_type, input_type, output_type):
+    if isinstance(output_type, list):
```

Comment:
Please update https://github.com/ibis-project/ibis/blob/master/ibis/expr/operations.py#L3462 to do proper rlz checking instead of checking it here.

See https://github.com/ibis-project/ibis/blob/master/ibis/expr/rules.py and other examples in https://github.com/ibis-project/ibis/blob/master/ibis/expr/operations.py

Reply:
Oops, this shouldn't be here (the checking is already done in UserDefinedFunction `__init__`)

Reply:
I'll look into doing checking using rlz

Reply:
I was trying rlz checking, but now I think it actually might be more appropriate to do the checking before we create the node (in other words, not using rlz)

Using rlz seems to be geared towards describing/validating each argument of a node, when the argument is in its "final form" in the node

**E.g.**
We want to make sure `output_type=[dt.double]` is caught and an error is raised. (Note that `output_type` "starts out" as a Python `list`)

So we can detect this syntax by (at some point) checking if the `output_type` is a Python `list`

But by the time we're in the node doing rlz checking, `output_type` would have been converted to an Ibis datatype [here](https://github.com/ibis-project/ibis/blob/master/ibis/udf/vectorized.py#L106). The rule would see `output_type` as an Ibis `Array`, which I think is valid (user could have intentionally specified `output_type=dt.Array(dt.double)`) so we can't raise an error

Reply:
There are workarounds that would let us still use a rlz check (e.g. do the type conversion in the node after validation has been done) but after exploring this I feel like it makes sense to conceptually treat the validation as checking the user's input to the UDF API as opposed to validating the node arguments

Reply:
(I still plan to remove this particular check in `_udf_decorator`, it's redundant (the checking is already done [elsewhere](https://github.com/ibis-project/ibis/blob/510f68d91c90d76e8958107765e383ca6088b46b/ibis/udf/vectorized.py#L28)))

Reply:
Done!